### ISSUE TITLE
NonEmptyAtclauseDescription does not support tokens property correctly

### ISSUE THREAD

**Issue by andrew-g-za** [checkstyle/checkstyle#3147]
When I attempt to customize the NonEmptyAtclauseDescription and select only specific javadoc tokens, the check fails.

NonEmptyAtClause.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="charset" value="UTF-8"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
		<property name="tokens" value="PARAM_LITERAL,RETURN_LITERAL,DEPRECATED_LITERAL" />
	</module>
  </module>
</module>
```

JavaDocTest.java:
```Java
public class JavaDocTest
{
	/**
	 * sample javadoc.
	 *
	 * @param p1
	 * 	Some param.
	 * @throws Exception
	 */
	public void javaDocFailsCheck(int p1) throws Exception
	{
		
	}
}
```

Command line:

```java -jar checkstyle-6.17-all.jar -c NonEmptyAtClause.xml JavaDocTest.java```

Result:

```
Exception in thread "main" java.lang.IllegalArgumentException: given name RETURN_LITERAL
        at com.puppycrawl.tools.checkstyle.utils.TokenUtils.getTokenId(TokenUtils.java:136)
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:219)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:162)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:138)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:176)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:138)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:326)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:159)
```

The 'tokens' property is generated by the eclipse-cs plugin:
 
![capture](https://cloud.githubusercontent.com/assets/11553559/14922029/93e5d576-0e35-11e6-969c-7cd79a68809e.JPG)

Ticking the 'Throws literal' option changes the configuration by removing the custom property (as all the default tokens are then selected) - at which point checkstyle runs without failing.

**Comment by romani** [comment]
please report issue like described here - http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug , all nuances do matter.

**Comment by andrew-g-za** [comment]
Sorry, i've updated the issue to be more complete.

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.html

This Check is javadoc Check, 'javadocTokens' property should be used to customize behaviour.
Unfortunately documentation is only creating, we need to find a reason of confusion to explain all users how to avoid such problems.

@baratali , pay attention to this case.

**Comment by rnveach** [comment]
The tokens are named wrong too. We don't have a `DEPRECATED_LITERAL` or `PARAM_LITERAL` token. The return literal is named `LITERAL_RETURN`. Also we have no customizable `tokens` (not talking about `javadocTokens`) for this check, as shown by the [documentation](http://checkstyle.sourceforge.net/config_javadoc.html#NonEmptyAtclauseDescription).

**Comment by romani** [comment]
this is problem of Eclipse-cs project 

PARAM_LITERAL is from JavadocTokenTypes not a TokenTpes.

http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html
http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html

generated xml config is incorrect.

But there is no way for now to override tokens https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java#L104
in Javadoc Checks.

**Comment by romani** [comment]
issue at eclipse-cs is opened - https://sourceforge.net/p/eclipse-cs/bugs/404/

I am closing this issue as actual fix will be at #2427 .

**Comment by andrew-g-za** [comment]
Many thanks.
